src/controllers/refreshTokenController.ts

```diff
@@ -0,0 +1,35 @@
+import { Response, Request } from 'express';
+import { Cache } from "../datasources/cache";
+import { refreshAuthTokens, setTokenCookie } from '../services/tokenService';
+import { generalConfig } from '../config/generalConfig';
+import { buildContext } from '../context';
+import { formatLogMessage, logger } from '../logger';
+
+export const refreshTokenController = async (req: Request, res: Response) => {
+  try {
+    const originalRefreshToken = req.cookies?.dmspr?.toString();
+
+    if (originalRefreshToken) {
+      const cache = Cache.getInstance();
+      const context = buildContext(logger, cache);
+      const { accessToken, refreshToken } = await refreshAuthTokens(cache, context, originalRefreshToken);
+
+      // If it successfully regenerated an access token
+      if (accessToken) {
```

Comment:
L18 - Do we need to check for both the accessToken and the refreshToken before setting cookies for them?

Reply:
yeah, that's probably a good idea